### PR TITLE
box mean: ensure correct behavior when radius exceeds half of image size

### LINKED ISSUE #8076: tone equalizer - weird artefacts and color shifts appear when zooming in
**Describe the bug**
I get weird artefacts and colour shifts when zooming in and out, it's quite random. And it only happens when the tone equalizer is turned on using the "guided filter". The actual weird shapes come from the mask which is generated by the guided filter.

The following (scene referred) modules are active: 

And this is how it could look like:
Screenshot from 2021-02-02 21-46-45

I also added a video, sorry for the zip, but unfortunately .webm as the one proper format for the web is not allowed with github :roll_eyes:
[Screencast from 02.02.2021 15:21:49.zip](https://github.com/darktable-org/darktable/files/5911816/Screencast.from.02.02.2021.15.21.49.zip)

This happens with OpenCL active/disabled (`--disable-opencl`), as well as SSE2 active/disabled (`--conf codepaths/sse2=false`).

**Steps to reproduce**
1. Choose a picture like https://discuss.pixls.us/t/that-day-at-route-1/23134
2. Download the following profile with a minimal set of active modules (sorry for the ZIP, github doesn't allow XML files): [DSC_0694.NEF.zip](https://github.com/darktable-org/darktable/files/5920220/DSC_0694.NEF.zip)

3. Zoom in at different positions in the picture and notice the weird artefacts

**Platform (please complete the following information):**
* Darktable Version: lastest master (3.5.0+913~g48425e51d-dirty)
* OS: Ubuntu 20.04

### PR DESCRIPTION
**Pull request by ralfbrown**
Hopefully fixes #8076 (I wasn't able to reproduce the colors/ghosting, but did get it to crash when zoomed in).

### CODE REVIEW

**Comment by da-phil** [comment]
For me it fixes https://github.com/darktable-org/darktable/issues/8076, thanks!

File: src/common/box_filters.c [hunk 1]
```diff
@@ -63,14 +63,14 @@ static void blur_horizontal_1ch(float *const restrict buf, const int height, con
...
-    for (int x = 0; x < radius && x < width ; x++)
+    for (int x = 0; x < MIN(radius,width) ; x++)
     {
       L += buf[index+x];
       hits++;
     }
     // process the blur up to the point where we start removing values
     int x;
-    for (x = 0; x <= radius && x < width; x++)
+    for (x = 0; x <= radius && x + radius < width; x++)
```
Comment (da-phil) [inline src/common/box_filters.c:73]: Although the compiler is probably doing the right thing on interpreting this expression, it is still complicated to read and therefore would propose to add some braces:
```suggestion
    for (x = 0; x <= radius && ((x + radius) < width); x++)
```

Comment (TurboGit) [reply src/common/box_filters.c:73]: Agreed, yet PR is an important fix. I'll merge and do the clean-up in a separate commit.

Comment (da-phil) [reply src/common/box_filters.c:73]: If you want I could also do that ;)

Comment (da-phil) [reply src/common/box_filters.c:73]: Ah, I saw you already did that on master, alright.

**Pull request merged**